src/Func.cpp

```diff
@@ -1090,6 +1090,36 @@ void Stage::split(const string &old, const string &outer, const string &inner, c
             << "Use TailStrategy::GuardWithIf instead.";
     }
 
+    bool predicate_loads_ok = !exact;
+    if (predicate_loads_ok) {
```

Comment:
How about `&& tail == TailStrategy::PredicateLoads` here to avoid all this logic if it's not the selected tail strategy.

Reply:
nice catch, thanks